test.js

```diff
@@ -6,13 +6,49 @@
 var leftPad = require("./");
 var test = require("tape");
 
-test('left pad', function (assert) {
-  assert.plan(7);
-  assert.strictEqual(leftPad('foo', 5), '  foo');
+test('edge cases', function (assert) {
+  assert.plan(12);
   assert.strictEqual(leftPad('foobar', 6), 'foobar');
+  assert.strictEqual(leftPad('foobar', 5), 'foobar');
+  assert.strictEqual(leftPad('foobar', -1), 'foobar');
+  assert.strictEqual(leftPad('foobar', 6, '1'), 'foobar');
+  assert.strictEqual(leftPad('foobar', 5, '1'), 'foobar');
+  assert.strictEqual(leftPad('foobar', -1, '1'), 'foobar');
+  assert.strictEqual(leftPad('foobar', 8, ''), '  foobar');
+  assert.strictEqual(leftPad('foobar', 8, false), '  foobar', 'false default to space');
+  //assert.strictEqual(leftPad('foobar', 8, []), '  foobar', '[] default to space');
+  //assert.strictEqual(leftPad('foobar', 8, {}), '  foobar', '{} default to space');
```

Comment:
@zehua I understand. I don't see a module should handle everything. users should understand the contract. If not, and if it's a problem, they might already have opened an issue with real-life scenarios :) But yeah it's nothing right or wrong here, just a different paradigm I believe 😄